pkg/controller/common/helpers_test.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import (
+	"testing"
+
+	igntypes "github.com/coreos/ignition/config/v2_2/types"
+)
+
+func TestValidateIgnition(t *testing.T) {
+	testIgnConfig := igntypes.Config{}
+
+	// Test that an empty ignition config returns nil
+	isValid := ValidateIgnition(testIgnConfig)
+	if isValid != nil {
+		t.Errorf("Expected an error. Got: %v", isValid)
+	}
+
+	// Test that an invalid ignition config returns and error
+	tempUser1 := igntypes.PasswdUser{Name: "core", SSHAuthorizedKeys: []igntypes.SSHAuthorizedKey{"5678", "abc"}}
+	testIgnConfig.Passwd.Users = []igntypes.PasswdUser{tempUser1}
+	isValid = ValidateIgnition(testIgnConfig)
+	if isValid == nil {
+		t.Errorf("Expected an error from invalid ignition config")
+	}
+
+	// Test that a valid ignition config returns nil
+	testIgnConfig.Ignition.Version = "2.0.0"
+	isValid = ValidateIgnition(testIgnConfig)
+	if isValid != nil {
```

Comment:
optional: This could just be `require.Nil(isValid)`

Reply:
done